### PR TITLE
feat: add checkVersion option for mobile:installApp

### PR DESCRIPTION
**Pull request by KazuCocoa**
https://github.com/appium/appium/issues/19754 for XCUITest.

`checkVersion` will compare the installed app version via `mobile:installApp` as same as the new session requst.

### CODE REVIEW

File: docs/reference/execute-methods.md [hunk 1]
```diff
@@ -167,6 +167,7 @@ Name | Type | Required | Description | Example
...
 strategy | string | no | One of possible app installation strategies on real devices. This argument is ignored on simulators. If not provided then the value of `appium:appInstallStrategy` is used. If the latter is also not provided then `serial` is used. See the description of `appium:appInstallStrategy` capability for more details on available values. | parallel
+checkVersion | bool | no | If set to `true`, it will make xcuitest driver to verify whether the app version currently installed on the device under test is older than the one, which is provided as `app` value. No app reinstall is going to happen if the candidate app has the same or older version number than the already installed copy of it. The version number used for comparison must be provided as [CFBundleVersion](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion) [Semantic Versioning](https://semver.org/)-compatible value in the application's `Info.plist`. No validation is performed and the `app` is installed if `checkVersion` was not provided or `false`, which is default behavior. | true
```
Comment (mykola-mokhnach) [inline docs/reference/execute-methods.md:170]: I assume if one would try to install an older (or same) version of an existing app without uninstalling it first then an installation error will be thrown by the system.
Same with Android

File: lib/driver.js [hunk 2]
```diff
@@ -1588,9 +1588,8 @@ class XCUITestDriver extends BaseDriver {
     return true;
   }
 
-  async checkAutInstallationState() {
-    // @ts-expect-error - do not assign arbitrary properties to `this.opts`
-    const {enforceAppInstall, fullReset, noReset, bundleId, device, app} = this.opts;
+  async checkAutInstallationState(opts = {}) {
```
Comment (mykola-mokhnach) [inline lib/driver.js:1591]: could you please add typedefs to this method?

File: docs/reference/execute-methods.md (hunk 1 above)
Comment (KazuCocoa) [reply docs/reference/execute-methods.md:170]: Maybe possibly only the older ones could return an error. In a new session request, I saw an older one returned an error in a real device. The same one succeeded.

File: lib/driver.js [hunk 3]
```diff
@@ -1594,24 +1594,15 @@ class XCUITestDriver extends BaseDriver {
...
    * Check if the given app can be installed, or should uninstall before installing it.
    *
-   * @param {CheckAutInstallationStateOption} [opts]
+   * @param {XCUITestDriverOpts} [opts]
    * @returns {Promise<CheckAutInstallationState>}
    */
   async checkAutInstallationState(opts) {
-    const {enforceAppInstall, fullReset, noReset, bundleId, device, app} = opts ?? {};
+      // @ts-expect-error - do not assign arbitrary properties to `opts`
+      const {enforceAppInstall, fullReset, noReset, bundleId, device, app} = opts ?? {};
```
Comment (mykola-mokhnach) [inline lib/driver.js:1605]: extra indent here

File: lib/driver.js [hunk 4]
```diff
@@ -1588,9 +1588,21 @@ class XCUITestDriver extends BaseDriver {
     return true;
   }
 
-  async checkAutInstallationState() {
-    // @ts-expect-error - do not assign arbitrary properties to `this.opts`
-    const {enforceAppInstall, fullReset, noReset, bundleId, device, app} = this.opts;
+  /**
+   * @typedef {Object} CheckAutInstallationState
```
Comment (mykola-mokhnach) [inline lib/driver.js:1592]: CheckAutInstallationState -> AutInstallationState

Type names must not contain verbs

File: lib/driver.js [hunk 5]
```diff
@@ -1588,9 +1588,21 @@ class XCUITestDriver extends BaseDriver {
...
+   * @property {boolean} install
+   * @property {boolean} skipUninstall
+   * /
+
+  /**
+   * Check if the given app can be installed, or should uninstall before installing it.
+   *
+   * @param {XCUITestDriverOpts} [opts]
```
Comment (mykola-mokhnach) [inline lib/driver.js:1600]: Lets just only enumerate the options we consider in this method. Explicit is better than implciit

File: docs/reference/execute-methods.md (hunk 1 above)
Comment (KazuCocoa) [reply docs/reference/execute-methods.md]: For real devices, the CFBundleVersion/short version did not affect the installation. So any errors never occurred while installing to lower CFBundleVersion/newer CFBundleVersion.

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**